File: src/main/java/co/uk/jiveelection/campaign/injection/translator/TranslatorModule.java
package co.uk.jiveelection.campaign.injection.translator;

import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslatorFactory;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class TranslatorModule {

    @Provides
    @Singleton
    public static InMemoryJiveTranslatorFactory providesInMemoryJiveTranslatorFactory() {
        return new InMemoryJiveTranslatorFactory();
    }

}

File: src/main/kotlin/co/uk/jiveelection/campaign/injection/translator/TranslatorModule.kt
package co.uk.jiveelection.campaign.injection.translator

import co.uk.jiveelection.campaign.translator.memory.InMemoryJiveTranslatorFactory
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object TranslatorModule {

    @Provides
    @Singleton
    fun providesInMemoryJiveTranslatorFactory(): InMemoryJiveTranslatorFactory {
        return InMemoryJiveTranslatorFactory()
    }

}
